torch/cuda/memory.py

```diff
@@ -1010,24 +1029,21 @@ def use_count(self) -> int:
         r"""Returns the reference count of this pool."""
         return super().use_count()
 
+    def snapshot(self):
+        r"""Return a snapshot of the CUDA memory allocator pool state across all
+        devices.
 
-class MemPoolContext(_MemPoolContext):
-    r"""MemPoolContext holds the currently active pool and stashes the previous
-    pool. On deletion it makes the previous pool active.
-
-    Args:
-        pool(torch.cuda.MemPool): a MemPool object to be made active so that
-        allocations route to this pool.
-
-    """
+        Interpreting the output of this function requires familiarity with the
+        memory allocator internals.
 
-    def __init__(self, pool: MemPool):
-        super().__init__(pool)
-
-    @staticmethod
-    def active_pool() -> Optional[_MemPool]:
-        r"""Returns the active MemPool"""
-        return _MemPoolContext.active_pool()
+        .. note::
+            See :ref:`cuda-memory-management` for more details about GPU memory
+            management.
+        """
+        ctx = MemPoolContext(self)
+        snapshot = torch.cuda.memory_snapshot()
+        del ctx
```

Comment:
try..finally better in case memory_snapshot throws and something in backtrace keeps locals live

Reply:
Added the try-finally. Merging now.